accepted/2022/FunctionPointers.md

```diff
@@ -0,0 +1,706 @@
+- Pointers: `ElementType`, `IsPointer`, `IsPointerImpl`, `MakePointerType()`
+- Arrays: `ElementType`, `IsArray`, `IsSZArray`,  `IsVariableBoundArray`, `GetArrayRank()`, `IsArrayImpl()`, `MakeArrayType()`
+- Enums: `IsEnum`, `GetEnumName()`, `GetEnumNames()`, `GetEnumUnderlyingType()`, `GetEnumValues()`, `GetEnumValuesAsUnderlyingType()`, `IsEnumDefined()`
+- Generics: `ContainsGenericParameters`, `GenericParameterAttributes`, `GenericParameterPosition`, `GenericTypeArguments`, `IsConstructedGenericType`, `IsGenericMethodParameter`, `IsGenericParameter`, `IsGenericType`, `IsGenericTypeDefinition`, `IsGenericTypeParameter`, `GetGenericArguments()`, `GetGenericParameterConstraints()`, `GetGenericTypeDefinition()`, `MakeGenericMethodParameter()`, `MakeGenericSignatureType()`, `MakeGenericType()`
+
+The original design, however, did include a `FunctionPointerParameterInfo` that contained a mechanism to obtain both the type and custom modifiers for each parameter. The proposed design here no longer directly exposes custom modifiers for parameters, so the `FunctionPointerParameterInfo` is no longer necessary since the only state remaining is just the parameter type, which is now directly lifted to `Type` via `Type[] GetFunctionPointerParameterTypes()`.
+
+## Custom modifiers
+The [7.0 introspection attempt](https://github.com/dotnet/runtime/pull/71516) returned all custom modifiers including modifiers not used or needed by the runtime, such as parameter modifiers for `ref`, `out` and `in`. Due to concerns with that approach, the function pointer feature was pulled from late in 7.0 in order to properly vet and approve a new design (this design) for v8. The intent is to add this functionality early in v8 which will also allow time for feedback.
+
+Consider the following where `ref\out\in` all map to a reference (`&`) by the runtime:
+```cs
+class MethodHolder
+{
+    public static void M1(in int i) { }
+    public static void M2(ref int i) { }
+    public static void M3(out int i) { i = 42; }
+}
+
+Type t1 = typeof(MethodHolder).GetMethod("M1").GetParameters()[0].ParameterType;
+Type t2 = typeof(MethodHolder).GetMethod("M2").GetParameters()[0].ParameterType;
+Type t3 = typeof(MethodHolder).GetMethod("M3").GetParameters()[0].ParameterType;
+
+Debug.Assert(t1 == t2 && t2 == t3); // True today; all are "int reference type" (int&)
+```
+
+Also consider [ILGenerator.EmitCalli()](https://docs.microsoft.com/dotnet/api/system.reflection.emit.ilgenerator.emitcalli#system-reflection-emit-ilgenerator-emitcalli(system-reflection-emit-opcode-system-runtime-interopservices-callingconvention-system-type-system-type())) as a reference implementation which does not support modifiers:
```

Comment:
Nit: Fix all links to use `learn.microsoft.com` instead of `docs.microsoft.com`?